src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -320,32 +323,61 @@ object TezosDatabaseOperations extends LazyLogging {
     (writeBlocks(blocks) andThen writeAccountsCheckpoint(accountUpdates.map(_.asTuple))).transactionally
   }
 
+  /** Custom postgres profile for enabling `insertOrUpdateAll` */
+  trait CustomPostgresProfile extends ExPostgresProfile {
+    // Add back `capabilities.insertOrUpdate` to enable native `upsert` support; for postgres 9.5+
+    override protected def computeCapabilities: Set[Capability] =
+      super.computeCapabilities + JdbcCapabilities.insertOrUpdate
+  }
+
+  object CustomPostgresProfile extends CustomPostgresProfile
+
   /**
-    * Writes the baking rights to the database
+    * Upserts baking rights to the database
     * @param bakingRightsMap mapping of hash to bakingRights list
     */
-  def writeBakingRights(bakingRightsMap: Map[BlockHash, List[BakingRights]]): DBIO[Option[Int]] = {
-    logger.info("Writing baking rights to the DB...")
+  def upsertBakingRights(bakingRightsMap: Map[BlockHash, List[BakingRights]]): DBIO[Option[Int]] = {
```

Comment:
Sorry, I'm missing something here: do we expect to have more than one possible value for the same (level, delegate) pair?

Reply:
Or maybe is it just updating the block's reference for the baking right? Is that correct?

Reply:
the idea is that fetching future rights will be fetched outside of the main lorre loop, and also we have fetching rights in the lorre which can get block hash, so as we download future rights we need to update it with block hash later. Also it was mentioned that should be prepared to be extracted so it could be run separately.

Reply:
Ok, I think I see the point now